examples/maglev/src/main.rs

```diff
@@ -1,26 +1,151 @@
-#![feature(box_syntax)]
 extern crate fnv;
-extern crate getopts;
+#[macro_use]
+extern crate lazy_static;
 extern crate netbricks;
-extern crate rand;
-extern crate time;
 extern crate twox_hash;
-use self::nf::*;
-use netbricks::config::{basic_opts, read_matches};
-use netbricks::interface::*;
-use netbricks::operators::*;
-use netbricks::scheduler::*;
-use std::env;
+use fnv::FnvHasher;
+use netbricks::common::Result;
+use netbricks::config::load_config;
+use netbricks::interface::{PacketRx, PacketTx};
+use netbricks::operators::{Batch, ReceiveBatch};
+use netbricks::packets::ip::v4::Ipv4;
+use netbricks::packets::ip::{Flow, IpPacket};
+use netbricks::packets::{Ethernet, Packet, Tcp};
+use netbricks::runtime::Runtime;
+use netbricks::scheduler::Scheduler;
+use std::collections::HashMap;
 use std::fmt::Display;
-use std::process;
-use std::sync::Arc;
-use std::thread;
-use std::time::Duration;
-mod nf;
+use std::hash::{BuildHasher, BuildHasherDefault, Hash, Hasher};
+use std::net::{IpAddr, Ipv4Addr};
+use std::sync::{Arc, RwLock};
+use std::{mem, slice};
+use twox_hash::XxHash;
 
-const CONVERSION_FACTOR: f64 = 1000000000.;
+const ENTRY_NUM: usize = 65537;
 
-fn test<T, S>(ports: Vec<T>, sched: &mut S)
+type FnvHash = BuildHasherDefault<FnvHasher>;
+type XxHashFactory = BuildHasherDefault<XxHash>;
+
+lazy_static! {
+    static ref FLOW_CACHE: Arc<RwLock<HashMap<usize, usize, FnvHash>>> = {
+        let m = HashMap::<usize, usize, FnvHash>::with_hasher(Default::default());
+        Arc::new(RwLock::new(m))
+    };
+    static ref BACKENDS: Vec<String> = {
+        let mut v = Vec::new();
+        v.push("Larry".to_string());
+        v.push("Moe".to_string());
+        v.push("Curly".to_string());
+        v
+    };
+    static ref LUT: Maglev = Maglev::new(ENTRY_NUM);
+}
+
+trait Stamper {
+    fn stamp_flow(&mut self, dst_ip: usize) -> Result<()>;
+}
+
+impl<E: IpPacket> Stamper for Tcp<E> {
+    fn stamp_flow(&mut self, dst_ip: usize) -> Result<()> {
+        self.envelope_mut().set_dst(IpAddr::V4(Ipv4Addr::new(
```

Comment:
if you call `set_src_ip` on `Tcp` directly, it will recalc the checksum too. then you don't need to `cascade` after if that's the only change made.

Reply:
def.

Reply:
dst_ip, but yep